### ISSUE TITLE
Provider Configuration Inconsistencies

### ISSUE THREAD

**Issue by jmusits** [Snowflake-Labs/terraform-provider-snowflake#2139]
**Provider Version**

Attempted on: 0.68.2, 0.73.0, 0.74.0

**Terraform Version**

1.5.6

**Describe the bug**

I am currently seeing some inconsistencies when using this provider.

There are two different ways that credentials are provided to the provider. The first is via `~/.snowflake/config`, the second is via environment variables.

When running `terraform plan` while using `~/.snowflake/config` I get the messaging that no changes have been detected, as expected, however when using the environment variables, terraform complains of drift (specifically a snowflake storage integration with s3), and then further on I get the following error `Error: object does not exist or not authorized` for a Snowflake database that is there.

The infrastructure was initially provisioned using `~/.snowflake/config` to authenticate.

I need to be able to get this working with the environment variables for running terraform in CI/CD.

**Expected behavior**

I expect that regardless of what methodology is used to configure the provider, running `terraform plan` would result in the same plan.

**Code samples and commands**

Below are my configurations:

**WORKING WITH ~/.snowflake/config**

Provider block in `main.tf`

```
provider "snowflake" {
  profile = "default"
  role    = "ACCOUNTADMIN"
}
```

Contents of `~/.snowflake/config`
```
[default]
account='<REDACTED>'
user='<REDACTED>'
password='<REDACTED>'
role='ACCOUNTADMIN'
```

**NOT WORKING WITH ENVIRONMENT VARIABLES**

Provider block in `main.tf`

```
provider "snowflake" {
  role = "ACCOUNTADMIN"
}
```

Environment variables:
```
SNOWFLAKE_ACCOUNT="<REDACTED>"
SNOWFLAKE_USER="<REDACTED>"
SNOWFLAKE_ROLE="ACCOUNTADMIN"
SNOWFLAKE_PASSWORD="<REDACTED>"
```
**Additional context**

**Comment by Shanjohn** [comment]
We encounter the same problem after 0.74.0 provider was released

**Comment by adrian-montero-hkkd** [comment]
Same issue here, tried both ways, no luck.

**Comment by LeoTheGriff** [comment]
I have the same issue when running Terraform via a service account. My user and the service account both have ACCOUNTADMIN but it shows 2 changes when using the service account (whether set by env variables or config file).
I have reviewed permissions in detail, and I couldn't find a loose grant.
Workaround: downgrade to and enforce 0.73.0

**Comment by sfc-gh-swinkler** [comment]
Hello, and sorry for the issue. This has been fixed as of 0.75. Please let me know if you have any further issue with regards to provider configuration auth.
